lua/kubectl/views/init.lua

```diff
@@ -167,6 +167,39 @@ function M.UserCmd(args)
   end)
 end
 
+function M.set_and_open_pod_selector(kind, name, ns)
+  local pod_view = require("kubectl.views.pods")
+  if not kind or not name or not ns then
+    return pod_view.View()
+  end
+  local get_selectors = { "get", kind, name, "-n", ns, "-o", 'jsonpath="{.spec.selector.matchLabels}"' }
```

Comment:
A lot of magic going on here, it's probably super correct and all but what about doing a `-o json` then 

```
local pods = vim.decode.json(object)
local labels = {}
for pod in pods then
table.insert(labels, pod.selector.matchLabels
end
```
Wouldn't that save you the trouble of handling the jsonpath output?

Reply:
ok changed the code a bit. also found out about `vim.uri_encode()` which eliminates the need for a custom function :)))